packages/merge-on-green/src/merge-logic.ts

```diff
@@ -370,8 +372,9 @@ mergeOnGreen.statusesForRef = async function statusesForRef(
         `Looking for required checks in status checks for ${owner}/${repo}/${pr}.`
       );
       //since find function finds the value of the first element in the array, that will take care of the chronological order of the tests
-      const checkCompleted = checkStatus.find(
-        (element: CheckStatus) => element.context === check
+      const regexCheck = new RegExp(`^${check}`);
+      const checkCompleted = checkStatus.find((element: CheckStatus) =>
+        regexCheck.test(element.context)
```

Comment:
I've seen this message in the logs:

```
Cannot read property 'context' of undefined
```

Perhaps we need a filter/check on `element`?

Reply:
Hey! So I'm pretty sure that that message is caused by the 'iterateGetStatusi' function, which basically calls up to 10 pages of the status checks, even if they're not there. Currently, it's actually part of the functionality of the bot, but I agree we should have a better way of handling it. I created issue #487 to track it!